### PR TITLE
feat: let the token picker loads tokens from DB

### LINKED ISSUE #3931: [Demand]  Let the token picker loads tokens from DB
## Catalogue

- [ ] UI/UX
- [ ] Mask Core Feature
- [x] Mask Plugin Feature (Wallet)
- [ ] Security/Cryptography Feature
- [ ] Other
  - /_List it here_/

## Part 1: Introduction and Background

If the user have added a custom token on the dashboard side. The token picker should be able to list it.

## Notes

### PR DESCRIPTION
**Pull request by NikoManiac**
closes #3931

### CODE REVIEW

File: packages/web3-shared/src/hooks/useAssetsByTokenList.ts [hunk 1]
```diff
@@ -39,7 +46,7 @@ export function useAssetsByTokenList(tokens: FungibleTokenDetailed[]) {
             }
             return asset
         })
-        .sort(makeSortAssertFn(chainId, { isMaskBoost: true }))
+        .filter((m) => tokens.find((x) => isSameAddress(x.address, m.token.address)))
```
Comment (guanbinrui) [inline packages/web3-shared/src/hooks/useAssetsByTokenList.ts:49]:
```suggestion
        .filter((m) => tokens.find((x) => isSameAddress(currySameAddress(m.token.address))))
```

File: packages/maskbook/src/extension/options-page/DashboardComponents/FixedTokenList.tsx [hunk 2]
```diff
@@ -82,7 +85,9 @@ export function FixedTokenList(props: FixedTokenListProps) {
             ? renderTokens
                   .sort(makeSortTokenFn(chainId, { isMaskBoost: true }))
                   .map((token) => ({ token: token, balance: null }))
-            : assets
+            : !!keyword
+            ? assets
+            : assets.sort(makeSortAssertFn(chainId, { isMaskBoost: true }))
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/options-page/DashboardComponents/FixedTokenList.tsx:90]: It isn't a wise choice to sort on external variable directly.

File: packages/web3-shared/src/hooks/useERC20TokensDetailedFromTokenLists.ts [hunk 3]
```diff
@@ -11,40 +11,42 @@ import type { ERC20TokenDetailed } from '../types'
...
         async () => (!lists || lists.length === 0 ? [] : fetchERC20TokensFromTokenLists(lists, chainId)),
         [chainId, lists?.sort().join()],
     )
     //#endregion
-
     //#region fuse
     const fuse = useMemo(
         () =>
-            new Fuse(allTokens, {
+            new Fuse([...additionalTokens, ...tokensFromList], {
```
Comment (guanbinrui) [inline packages/web3-shared/src/hooks/useERC20TokensDetailedFromTokenLists.ts:27]: What will happen if it got the same token both from token lists and DB?

**Comment by wenluomask** [comment]
Checked.

File: packages/web3-shared/src/hooks/useERC20TokensDetailedFromTokenLists.ts (hunk 3 above)
Comment (NikoManiac) [reply packages/web3-shared/src/hooks/useERC20TokensDetailedFromTokenLists.ts:27]: It's OK.
```
 const renderTokens = uniqBy([...tokens, ...filteredTokens, ...(searchedToken ? [searchedToken] : [])], (x) =>
        x.address.toLowerCase(),
    )
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**